plugins/common/tls/config.go

```diff
@@ -57,9 +58,21 @@ func (c *ClientConfig) TLSConfig() (*tls.Config, error) {
 		return nil, nil
 	}
 
+	// TLS1.0 and TLS1.1 are effectively EOL, so make TLS1.2 the minimum. See
+	// https://github.com/influxdata/telegraf/issues/8171#issuecomment-777120753
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: c.InsecureSkipVerify,
 		Renegotiation:      tls.RenegotiateNever,
+		MinVersion:         tls.VersionTLS12,
```

Comment:
I like this idea, but we support a lot of really old software and this will likely be a breaking change for some of them. I'd like to manage this by putting this one line in its own PR, and when they don't specify the version spit out a message saying something like: `TLS 1.0 is deprecated and will no longer be the default in a future version of Telegraf. To suppress this message specify the minimum TLS version explicitly, eg tls_min_version = "TLS13"`. ... alternatively maybe we could find a way to warn when we establish a connection with a really old TLS version.. though that seems more complicated.

There's also a MaxVersion we might want to support, though I don't currently have a use case for it.